90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -1012,39 +1123,22 @@ load_be_cmdline() {
   if [ -e "${BASE}/noresume" ]; then
     zdebug "${BASE}/noresume set, processing ${zfsbe_args}"
     # Must replace resume= arguments and append a noresume
-    zfsbe_args="$( awk <<< "${zfsbe_args}" '
-      BEGIN {
-        quot = 0;
-        supp = 0;
-        ORS = " ";
-      }
-
-      {
-        for (i=1; i <= NF; i++) {
-          if ( quot == 0 ) {
-            # If unquoted, determine if output should be suppressed
-            if ( $(i) ~ /^resume=/ ) {
-              # Argument starts with "resume=", suppress
-              supp = 1;
-            } else {
-              # Nothing else is suppressed
-              supp = 0;
-            }
-          }
+    zfsbe_args="$( suppress_kcl_arg resume "${zfsbe_args}" ) noresume"
+  fi
 
-          # If output is not suppressed, print the field
-          if ( supp == 0 && length($(i)) > 0 ) {
-            print $(i);
-          }
+  # shellcheck disable=SC2154
+  if [ "${zbm_set_hostid}" -eq 1 ]; then
+    # Try to assume a matched hostid if none is defined
+    if [ -z "${spl_hostid}" ] && [ -r "${BASE}/spl_hostid" ]; then
+      read -r spl_hostid < "${BASE}/spl_hostid"
+    fi
 
-          # If an odd number of quotes are in this field, toggle quoting
-          if ( gsub(/"/, "\"", $(i)) % 2 == 1 ) {
-            quot = (quot + 1) % 2;
-          }
-        }
-        printf "noresume";
-      }
-    ' )"
+    # Only override hostid arguments if one can be provided
+    if [ -n "${spl_hostid}" ]; then
+      zdebug "overriding spl_hostid and spl.spl_hostid in: ${zfsbe_args}"
+      zfsbe_args="$( suppress_kcl_arg spl_hostid "${zfsbe_args}" )"
+      zfsbe_args="$( suppress_kcl_arg spl.spl_hostid "${zfsbe_args}" ) spl.spl_hostid=0x${spl_hostid}"
```

Comment:
I put this at the end just in case we miss something somehow; I think the kernel prefers the last value.

Reply:
Good call.